### PR TITLE
Implement subConfigKeys() for Config

### PR DESCRIPTION
**Pull request by VovaStelmashchuk**
Implementation the subConfigs method for Config.

I start implementation of subConfigs methods for Config interface children. Any suggestion or review welcome, I will add more tests by myself in few days.

@BraisGabin in case you have some existing work on the task fill free to close the PR and cherry pick changes if you needed.

Part of #6738

### CODE REVIEW

**Review by BraisGabin — COMMENTED** [review]
Thanks for the PR :)

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt [hunk 1]
```diff
@@ -20,6 +20,11 @@ interface Config {
      */
     fun subConfig(key: String): Config
 
+    /**
+     * Returns a map of all sub configurations.
+     */
+    fun subConfigs(): Map<String, Config>
```
[the remark this answers is not in the document]
Comment (VovaStelmashchuk) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt:26]: I like the idea about add method which returns a Set<String> , the set of keys. 

I do not agree with extension method Config.subConfigsMap in first iteration. Looks like we will use the subconfigs itself only in Analyzer, the associated keys with subconfigs will be done in the Analyzer. I will rewrite the implemenation to returns the Set today.

[the remark this answers is not in the document]
Comment (BraisGabin) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt:26]: Agree. If needed, we can have the extension as a private extension in `Analyzer`. But no need to pollute `detekt-api`.

**Comment by VovaStelmashchuk** [comment]
The pr ready for review. Pls ignore the CodeCov report at the moment, the method subConfigKeys will be covered by tests in case we start using the API.

**Review by BraisGabin — APPROVED** [review]
This looks good! Now I'm thinking how to differentiate between subconfigs and normal values. Example:

```yaml
Style:
  Excludes: [**]
  MagicNumber:
    Active: true
```

How can I know that `excludes` is not a rule?

**Comment by VovaStelmashchuk** [comment]
I think we will filter the keys in Analyzer. Something like this.

```kotlin
val ruleNames ruleSet.rules.map { it::javaclass.classname }
val ruleKeys = key.filters { ruleNames.startWith(it) }  
```

**Pull request merged**